packages/ember-routing/lib/system/router.js

```diff
@@ -705,6 +706,24 @@ var EmberRouter = EmberObject.extend(Evented, {
       run.cancel(this._slowTransitionTimer);
     }
     this._slowTransitionTimer = null;
+  },
+
+  _handledErrors: computed(function() {
```

Comment:
Directly branding the error seems ok to me (it really isn't a big change for this PR as the basic infrastructure to brand is done), but I believe that we have a few tests in Ember itself that `reject("asdf")` or `throw "asdf"` which will need to be changed.